qe/drivers.py

```diff
@@ -1884,37 +1854,35 @@ def dispatch_psi(comm, psi_i):
     :param psi_i: Psi_det, list of determinants"""
 
     rank = comm.Get_rank()  # Rank of current process
-    size = comm.Get_size()  # No. of processes currently running
+    world_size = comm.Get_size()  # No. of processes currently running
+    MPI_master_rank = 0  # Denote master rank
 
     if rank == 0:
         indices = np.arange(len(psi_i), dtype="i")
-        number_of_dets, res = divmod(len(psi_i), size)
-        # Number of determinants sent to each worker; First `res` processes sent one extra
+        number_of_dets, res = divmod(len(psi_i), world_size)
+        # Number of determinants sent to each rank; First `res` processes sent one extra
         sendcounts = np.array(
-            [number_of_dets + 1 if i < res else number_of_dets for i in range(size)]
+            [number_of_dets + 1 if i < res else number_of_dets for i in range(world_size)],
+            dtype="i",
         )
         displacement = np.array(
-            [sum(sendcounts[:i]) for i in range(size)]
-        )  # Entry i of displacement vector is the index of the first determinant process i receives
+            [sum(sendcounts[:i]) for i in range(world_size)], dtype="i"
```

Comment:
Maybe you can use np.sum + axis (https://numpy.org/doc/stable/reference/generated/numpy.sum.html#numpy-sum)

Reply:
Can you elaborate ?

Reply:
`numpy.sum(sendcounts, axis=2)` may be what you are doing

Reply:
I think this sums along a particular array dimension, I'm summing up to a particular index.

Reply:
I think you're looking for `accumulate`